python/ray/experimental/channel.py

```diff
@@ -74,47 +101,131 @@ def __init__(
         Returns:
             Channel: A wrapper around ray.ObjectRef.
         """
-        if buffer_size_bytes is None:
-            if _base_ref is None:
-                raise ValueError(
-                    "One of `buffer_size_bytes` or `_base_ref` must be provided"
-                )
-            self._base_ref = _base_ref
-        else:
+        is_creator = False
+        assert len(readers) > 0
+
+        if _writer_ref is None:
             if not isinstance(buffer_size_bytes, int):
                 raise ValueError("buffer_size_bytes must be an integer")
-            self._base_ref = _create_channel_ref(buffer_size_bytes)
 
-        if not isinstance(num_readers, int):
-            raise ValueError("num_readers must be an integer")
+            self._writer_node_id = (
+                ray.runtime_context.get_runtime_context().get_node_id()
+            )
+            self._writer_ref = _create_channel_ref(self, buffer_size_bytes)
+
+            if readers[0] is None:
+                # Reader and writer are on the same node.
+                self._reader_node_id = self._writer_node_id
+                self._reader_ref = self._writer_ref
+            else:
+                # Reader and writer are on different nodes.
+                fn = readers[0].__ray_call__
+                self._reader_node_id = ray.get(fn.remote(_get_node_id))
```

Comment:
It'd be good to call this on each reader and check that the node IDs match. Otherwise, channel creation may fail silently if readers are on different nodes. For now, you can just throw a NotImplementedError if the readers are on different nodes.

Reply:
Done.